pkg/controller/utils/utils.go

```diff
@@ -107,12 +112,41 @@ func CheckCloudCredCreation(awsClient ccaws.Client, logger log.FieldLogger) (boo
 	return CheckPermissionsAgainstActions(awsClient, credMintingActions, logger)
 }
 
+// getClientDetails will return the *iam.User associated with the passed in credentials,
+// a boolean indicating whether the user is the 'root' account, and any error encountered
+// while trying to gather the info
+func getClientDetails(awsClient ccaws.Client) (*iam.User, bool, error) {
+	rootUser := false
+
+	user, err := awsClient.GetUser(nil)
+	if err != nil {
+		return nil, rootUser, fmt.Errorf("error querying username: %v", err)
+	}
+
+	// Detect whether the AWS account's root user is being used (unsupported)
```

Comment:
why "unsupported"?  AWS recommends (wisely) that you not do this, but the cluster should come up fine when launched by the account's root.  And if there are voids-your-warranty type issues vs. Red Hat support for such clusters, then we probably want to surface that at a higher-level than a library comment ;).

Reply:
that was earlier today when i thought we were going to error out in the face of the root user